src/EditorFeatures/Core/GoToDefinition/ISymbolicNavigationService.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    internal interface ISymbolicNavigationService : ILanguageService
+    {
+        /// <summary>
+        /// Is this a symbol of your language? If it is, tell us and you can take over the navigation however you would like.
+        /// </summary>
+        Task<bool> TryNavigateToSymbol(ISymbol symbol, CancellationToken cancellationToken);
```

Comment:
i would also have an overload that takes in a fully qualified *metadata* name for the symbol.  And, maybe that should just be the entirety of hte API.  It feels, to me, that you could only navigate to some top-level symbol in another language.  So it would always be a symbol that you only know about from metadata.  And if it's from metadata, you'd have a full metadata name for it.